File: packages/@simpli/cli-scaffold/generator/template/src/model/response/TagResp.ts
/**
 * TagResp
 * @author martinlabs CRUD generator
 */
import {ID, TAG, Resource} from '@/simpli'
import {ResponseSerialize} from '@/simpli'

import Tag from '@/model/Tag'
import Principal from '@/model/Principal'

export default class TagResp extends Resource {
  readonly $endpoint: string = '/Crud/Tag{/id}'

  @ResponseSerialize(Tag)
  tag: Tag = new Tag()

  @ResponseSerialize(Principal)
  allPrincipal: Principal[] = []

  get $id() {
    return this.tag.$id
  }

  set $id(val: ID) {
    this.tag.$id = val
  }

  get $tag() {
    return this.tag.$tag
  }

  set $tag(val: TAG) {
    this.tag.$tag = val
  }
}

File: packages/@simpli/cli-server/generator/template/src/main/java/br/com/martinlabs/usecase/crud/response/TagResp.kt
package br.com.martinlabs.usecase.crud.response

import io.swagger.annotations.ApiModel
import io.swagger.annotations.ApiModelProperty
import br.com.martinlabs.usecase.model.Tag
import br.com.martinlabs.usecase.model.Principal

@ApiModel(value = "TagResp")
class TagResp {

    var tag: Tag? = null

    @ApiModelProperty(value = "Possible Principal values")
    var allPrincipal: MutableList<Principal>? = null
}
